dashboard/app/models/pd/application/teacher_application_base.rb

```diff
@@ -65,6 +64,15 @@ def update_user_school_info!
       end
     end
 
+    def update_scholarship_status(scholarship_status)
+      scholarship_info = Pd::ScholarshipInfo.find_by(user: user, application_year: application_year) || Pd::ScholarshipInfo.new(user: user)
```

Comment:
should this be using `find_or_create_by`?

Reply:
I don't think that will work here. The `scholarship_status` field is required to create a ScholarshipInfo, so I would need to pass that in, but if there is an existing one, it will have an old status and wouldn't be found by that statement.